ttnctl/cmd/devices_set.go

```diff
@@ -153,13 +153,17 @@ var devicesSetCmd = &cobra.Command{
 			dev.Description = in
 		}
 
-		if in, err := cmd.Flags().GetStringArray("customs-keys"); err == nil && len(in) > 0 {
+		if in, err := cmd.Flags().GetStringArray("custom-attr"); err == nil && len(in) > 0 {
 			for _, v := range in {
 				s := strings.SplitN(v, ":", 2)
-				if dev.CustomKeys == nil {
-					dev.CustomKeys = make(map[string]string, len(in))
+				if dev.Attributes == nil {
+					dev.Attributes = make(map[string]string, len(in))
+				}
+				if len(s) == 2 {
+					dev.Attributes[s[0]] = s[1]
+				} else {
+					dev.Attributes[s[0]] = ""
```

Comment:
It's not. It's to protect from panic if someone add a key without value, the result will be an empty value for a key in the map.

Removing a custom attributes is currently not possible. Thinking about way to do it without adding more to  flags